cellmlmanip/model.py

```diff
@@ -137,11 +141,20 @@ def _find_unit_info(self, expr):
         return None
 
     def find_variable(self, search_dict):
-        """Finds variables in this model that match the given key: value pairs
+        """Finds variables in this model that fulfil the search criteria.
 
-        :param search_dict:
-        :return: a dictionary of key: value pairs
+        Each model variable has a number of attributes e.g. name, public_interface, assignment,
+        cmeta:id etc. Search criteria are described as key-value pairs in a dictionary. Those
+        model variables that fulfil _all_ key-value pairs in the search_dict are returned.
+
+        :param search_dict: a dictionary providing the search criteria
+        :return: a list of variables in the model matching the search criteria
         """
+
+        # throw error if empty search criteria given
+        if search_dict is None or not search_dict:
+            raise RuntimeError("Search criteria cannot be empty.")
```

Comment:
`ValueError` might be more appropriate here; I think that's typically used for an invalid argument to a function.